src/cli.js

```diff
@@ -0,0 +1,1601 @@
+import chalk from 'chalk';
+import fs from 'fs-extra';
+import { program, Command, Option } from 'commander';
+import { basename, dirname, join } from 'node:path';
+import { fileURLToPath, pathToFileURL } from 'node:url';
+import { unique } from './util/unique.js';
+import { ticonfig } from './util/ticonfig.js';
+import { initSDK } from './util/tisdk.js';
+import { expand } from './util/expand.js';
+import { arrayify } from './util/arrayify.js';
+import * as version from './util/version.js';
+import { Logger } from './util/logger.js';
+import { capitalize } from './util/capitalize.js';
+import wrapAnsi from 'wrap-ansi';
+import { TiError } from './util/tierror.js';
+import { prompt } from './util/prompt.js';
+import { applyCommandConfig } from './util/apply-command-config.js';
+import { TiHelp } from './util/tihelp.js';
+
+const { blue, bold, cyan, gray, green, magenta, red, yellow } = chalk;
+
+/**
+ * A curated list of built-in CLI commands and CLI commands defined in a
+ * Titanium SDK. These are hard coded for speed.
+ */
+const commands = {
+	config:  'get and set config options',
+	info:    'display development environment information',
+	module:  'displays installed Titanium modules',
+	sdk:     'manages installed Titanium SDKs',
+	setup:   'sets up the Titanium CLI'
+};
+
+const sdkCommands = {
+	build:   'builds a project',
+	clean:   'removes previous build directories',
+	create:  'creates a new project',
+	project: 'get and set tiapp.xml settings'
+};
+
+/**
+ * The Titanium CLI v5 requires the `--sdk <version>` to equal the
+ * `<sdk-version>` in the `tiapp.xml`. If they don't match, node-titanium-sdk's
+ * `ti.validateCorrectSDK()` will spawn a new Titanium CLI process with the
+ * correct `--sdk`. Due to the design of the Titanium CLI, this
+ * `GracefullyShutdown` error was thrown as an easy way to stop validating and
+ * skip executing the command.
+ *
+ * Since this Titanium CLI shim will ALWAYS match the `<sdk-version>` in the
+ * `tiapp.xml`, this really isn't used, but just in case, we'll define it and
+ * set it on the `CLI` instance.
+ */
+class GracefulShutdown extends Error {}
+
+process.setMaxListeners(666);
```

Comment:
Any reason for this amount of listeners?

Reply:
Node.js prints a warning if more than 10 listeners are added to an EventEmitter and the CLI plugin system uses an EventEmitter under the hood and 10 is not enough. Plus this line has been in the CLI for 11 years :) https://github.com/tidev/titanium-cli/blob/master/lib/titanium.js#L158